gensim/models/ensemblelda.py

```diff
@@ -0,0 +1,1268 @@
+                '{}% of tokens, which might be too much'.format(threshold, percent))
+        else:
+            logger.info('The given threshold of {} covered on average {}% of tokens'.format(threshold, percent))
+
+        return distances
+
+    def generate_topic_clusters(self, eps=0.1, min_samples=None):
+        """Runs the DBSCAN algorithm on all the detected topics from
+        the models in the ensemble and labels them with label-indices.
+
+        The final approval and generation of stable topics is done in
+        generate_stable_topics().
+
+        Parameters
+        ----------
+        eps : float
+            eps is 0.1 by default.
+        min_samples : number
+            min_samples is int(self.num_models / 2)
+        """
+
+        if min_samples is None:
+            min_samples = int(self.num_models / 2)
+
+        logger.info("Fitting the clustering model")
+
+        self.cluster_model = CBDBSCAN(eps=eps, min_samples=min_samples)
+        self.cluster_model.fit(self.asymmetric_distance_matrix)
+
+    def generate_stable_topics(self, min_cores=None):
+        """generates stable topics out of the clusters.
+        This function is the last step that has to be done
+        in the ensemble.
+
+        Stable topics can be retreived afterwards using
+        get_topics().
+
+        Parameters
+        ----------
+        min_cores : number
+            how many cores a cluster has to have,
+            to be treated as stable topic. That means, how many topics
+            that look similar have to be present, so that the average
+            topic in those is used as stable topic.
+
+            defaults to min_cores = min(3, max(1, int(self.num_models /4 +1)))
+
+        """
+
+        logger.info("Generating stable topics")
+
+        # min_cores being 0 makes no sense. there has to be a core for a cluster
+        # or there is no cluster
+        if min_cores == 0:
+            min_cores = 1
+
+        if min_cores is None:
+            # min_cores is a number between 1 and 3, depending on the number of models
+            min_cores = min(3, max(1, int(self.num_models / 4 + 1)))
+
+        results = self.cluster_model.results
+
+        # first, group all the learned cores based on the label,
+        # which was assigned in the cluster_model. The result is a
+        # dict of {group: [topic, ...]}
+        grouped_by_labels = {}
+        for topic in results:
+            if topic["is_core"]:
+                topic = topic.copy()
+
+                # counts how many different labels a core has as parents
+                topic["amount_parent_labels"] = len(topic["parent_labels"])
+
+                label = topic["label"]
+                if label not in grouped_by_labels:
+                    grouped_by_labels[label] = []
+                grouped_by_labels[label].append(topic)
+
+        # then aggregate their amount_parent_labels by maxing and parent_labels by concatenating
```

Comment:
This code block could be another separate function: e.g. _aggregate_grouped_labels.